lib/muser_ctx.c

```diff
@@ -2181,6 +2037,25 @@ process_request(lm_ctx_t *lm_ctx)
         return ret;
     }
 
+    /*
+     * TODO from now on if an error occurs we still need to reply. Move this
+     * code into a separate function so that we don't have to use goto.
+     */
+
+    hdr.msg_size -= sizeof(hdr);
+    if (hdr.msg_size > 0) {
+        cmd_data = malloc(hdr.msg_size);
+        if (cmd_data == NULL) {
+            ret = -ENOMEM;
+            goto reply;
+        }
+        ret = recv(lm_ctx->conn_fd, cmd_data, hdr.msg_size, 0);
+        if (ret < 0) {
+            ret = -errno;
+            goto reply;
+        }
+    }
+
     /* FIXME in most of the following function we check that hdr.count is >=
      * than the command-specific struct and there is an additional recv(2) for
```

Comment:
want to update the recv part of this?

Reply:
and does it even make sense try to move the size checks here themselves?

Reply:
> want to update the recv part of this?

Not sure I understand?

> and does it even make sense try to move the size checks here themselves?

Maybe, I'm tried to understand how we can further simplify code, I need to think about it a bit more and then come up with a suggestion.

Reply:
The FIXME is talking about the additional recv(2), but aren't these changes implementing exactly that?

Reply:
Correct, I removed the FIXME comment in a new PR.